src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -107,6 +109,7 @@ public ODataUtf8JsonWriter(Stream outputStream, bool isIeee754Compatible, Encodi
             this.bufferFlushThreshold = 0.9f * this.bufferSize;
             this.leaveStreamOpen = leaveStreamOpen;
             this.outputEncoding = encoding;
+            this.encoder = encoder ?? JavaScriptEncoder.Default;

```

Comment:
Pass `this.encoder` to the `Utf8JsonWriter` constructor below.

Reply:
I think it's already being passed.

Reply:
This is what is currently happening:

```c#
new JsonWriterOptions {
 ...
  Encoder == encoder ?? JavaScriptEncoder.Default
```

And this is what I'm proposing

```c#
new JsonWriterOptions {
 ...
  Encoder == this.encoder
```

While they're technically equivalent, the first one unnecessarily evaluates the `encoder ?? JavaScriptEncoder.Default` again and it is more prone to bugs. For example if we ever change the `this.encoder` to something else, we might forget to also make a similar change to the instance passed to the constructor and introduce bugs.